client/my-sites/subscribers/mutations/use-subscriber-remove-mutation.ts

```diff
@@ -113,8 +114,11 @@ const useSubscriberRemoveMutation = (
 						// Verify the response indicates successful deletion
 						emailRemoved = response?.deleted === true;
 					} catch ( e ) {
-						// Only throw if we haven't successfully removed them through any other method.
-						if ( ! wpcomRemoved ) {
+						// Consider "not_following" as a successful removal since they're already not following
+						if ( ( e as ApiResponseError )?.error === 'not_following' ) {
```

Comment:
I didn't see an impact from this check but it seems like something we should check for.